src/storm-pars/transformer/ParametricTransformer.cpp

```diff
@@ -0,0 +1,111 @@
+#include "storm-pars/transformer/ParametricTransformer.h"
+#include "storm/storage/sparse/ModelComponents.h"
+#include "storm/exceptions/IllegalArgumentException.h"
+#include "storm/exceptions/NotImplementedException.h"
+#include "storm/storage/SparseMatrix.h"
+#include "storm/models/sparse/StateLabeling.h"
+
+namespace storm {
+namespace transformer {
+
+std::shared_ptr<storm::models::sparse::Dtmc<storm::RationalFunction>> makeRewardsConstant(std::shared_ptr<storm::models::sparse::Dtmc<storm::RationalFunction>> const& pMC) {
+    STORM_LOG_THROW(pMC->hasUniqueRewardModel(), storm::exceptions::IllegalArgumentException, "pMC needs to have an unique reward model");
+    storm::storage::sparse::ModelComponents<storm::RationalFunction> modelComponents;
+
+    uint64_t nrStates = pMC->getTransitionMatrix().getColumnCount();
+    uint_fast64_t nrOfNewStates = 0;
+    auto rewardModel = pMC->getUniqueRewardModel();
+    for (uint64_t state = 0; state < nrStates; ++state) {
+        if (rewardModel.hasStateActionRewards() && !rewardModel.getStateActionReward(state).isConstant()) {
+            nrOfNewStates = nrOfNewStates + 2;
+        } else if (rewardModel.hasStateRewards() && !rewardModel.getStateReward(state).isConstant()) {
+            nrOfNewStates = nrOfNewStates + 2;
+        }
+    }
+
+    storm::storage::SparseMatrixBuilder<storm::RationalFunction> smb(nrStates + nrOfNewStates, nrStates + nrOfNewStates, 0, true);
+
+    uint64_t offset = 0;
+    std::vector<storm::RationalFunction> stateRewards(nrStates + nrOfNewStates, storm::RationalFunction(0));
+    storm::models::sparse::StateLabeling stateLabeling(nrStates + nrOfNewStates);
+    STORM_LOG_THROW(!pMC->getOptionalStateValuations(), storm::exceptions::NotImplementedException, "Keeping rewards constant while having state valuations is not implemented");
+    for (uint64_t state = 0; state < nrStates; ++state) {
+        storm::RationalFunction reward = storm::RationalFunction(0);
+        if (rewardModel.hasStateActionRewards()) {
+            reward = rewardModel.getStateActionReward(state);
+        } else {
+            STORM_LOG_ASSERT(rewardModel.hasStateRewards(), "Expecting model to have either state rewards or state action rewards");
+            reward = rewardModel.getStateReward(state);
+        }
+        for (auto& label : pMC->getStateLabeling().getLabelsOfState(state)) {
+            if (!stateLabeling.containsLabel(label)) {
+                stateLabeling.addLabel(label);
+            }
+            stateLabeling.addLabelToState(label, state + offset);
+        }
+
+        if (!reward.isConstant()) {
+            /*
+                         * We want to transform a state s with parametric reward r to 3 states, s0, s1 and s2.
+                         * s1 and s2 have as successor states the successor states of s
```

Comment:
Could we instead have a transition with prob 1 from s1 to s2, and then only have outgoing transitions from s2? That would reduce the number of  transitions?